### ISSUE TITLE
Refering to a parameter/local var in an initializer of a {.global.} var results in invalid C code

### ISSUE THREAD

**Issue by recloser** [nim-lang/Nim#9333]
compiling
```nim
proc foo(arg: int) =
    var g {.global.} = arg + 1
    echo g

foo(123)
```
produces
```
Error: execution of an external compiler program 'gcc -c  -w  -I'~/.choosenim/toolchains/nim-#devel/lib' -o ~/.cache/nim/bug_d/bug.c.o ~/.cache/nim/bug_d/bug.c' failed with exit code: 1

~/.cache/nim/bug_d/bug.c: In function 'NimMainModule':
~/.cache/nim/bug_d/bug.c:168:41: error: 'arg' undeclared (first use in this function)
  TM_rmH9bpP9b3ZIjFP4QhigizlA_2 = addInt(arg, ((NI) 1));
                                         ^~~
~/.cache/nim/bug_d/bug.c:168:41: note: each undeclared identifier is reported only once for each function it appears in
```

**Comment by alaviss** [comment]
If I understand this correctly, you want `g` to be initialized on the first call only, so you'd have to do this:

```nim
proc foo(arg: int) =
    var g {.global.} = 0
    once:
        g = arg + 1
    echo g
```

This is because [`{.global.}`][0]s are initialized during program startup, which of course, `arg` wouldn't be available then since it only exists within the `proc`.

The compiler should raise a proper error for this case however.

[0]: http://nim-lang.github.io/Nim/manual.html#pragmas-global-pragma

**Comment by Araq** [comment]
Duplicate of https://github.com/nim-lang/Nim/issues/3505
